.travis/runpylint.sh

```diff
@@ -17,6 +17,7 @@ set -e
 ME=$(basename $0)
 SCRIPTDIR=$(readlink -f $(dirname $0))
 
+. ${SCRIPTDIR}/utils.sh
```

Comment:
This change does not seem to make sense, since nothing is using utils.sh in this file.

Reply:
I would have to check the python script custom_pylint.py to see if it is using any environment variables defined in utils.sh to be sure (I don't think so) - so this is primarily for the sake of consistency - every script in `.travis/` includes `utils.sh` so that in the future we can control the common behavior of all scripts in one central location, and every script can rely on having a common set of environment variables/functions/behaviors.  Similarly with `config.sh` below.  I suppose your argument might be "well, we don't need it now, why not just add it in the future when we do need it?" and I suppose, yes, that seems like a reasonable argument, but for the sake of consistency, why not just do it now?

Reply:
IMHO it is bad style to have unnecessary code because it makes it harder to understand the code (in this case someone would need to check if utils.sh has an effect on runpylist.sh or not), makes it harder to change utils.sh (now more files needs to be checked if they might be broken by the changes) and it has a (probably low) performance impact. Static linters for other languages usually warn about unused imports, for example for Python. By not including unnecessary code, you also have consistency for a rule like everything includes only what it actually needs.

I will not block this PR if you have prefer to add this nonetheless.

Reply:
ok, now I remember - the function `lsr_setup_module_utils` is needed in order to run `pylint` in some repos like `storage` - so `runpylint.sh` must source both `utils.sh` in order to pick up the definition of the shell function `lsr_setup_module_utils`, and `config.sh` in order to run the shell function.  https://github.com/linux-system-roles/storage/blob/master/.travis/config.sh#L50

I guess this isn't a problem for network because you have your own way of solving this problem.

Reply:
I see. Is this good practice is shell programming? It seems bad to me that sourcing a file called `config.sh` is intended to have side effects by running code from a different file that is not even sourced in `config.sh`. IMHO cleaner approaches would be to

a)
set a flag in config.sh that indicates that the `lsr_setup_module_utils` function should be called and explicitly call where needed

or

b) 
make `config.sh` source `utils.sh` to get the `lsr_setup_module_utils` function (this would at least make it a little bit clearer that it is running code from `utils.sh`
(This does not solve the side effect of sourcing).

What do you think?

Reply:
with option b) - each script in `.travis/` would have the following:
```
. ${SCRIPTDIR}/utils.sh # OPTIONAL - only if the script requires the shell functions/env. vars.
. ${SCRIPTDIR}/config.sh # OPTIONAL - only if the script needs to allow the project to override behavior
```
so it may be that a script in `.travis/` will have both
```
. ${SCRIPTDIR}/utils.sh
. ${SCRIPTDIR}/config.sh
```
AND the project needs to override behavior in the script by calling a shell function defined in utils.sh:
```
${SCRIPTDIR}/config.sh:
. ${SCRIPTDIR}/utils.sh
lsr_setup_module_utils
```
So utils.sh will be sourced twice.  If this is what you intended, then this can work - I'll have to add "protection" to utils.sh so that it can either be sourced multiple times, or changed so that sourcing it a second or subsequent time will be a no-op.

Reply:
Actually (a) is my favorite, (b) is a compromise because it is still a little bit odd that the config contains code/has side effects. But, yes, a guarding mechanism would be needed.